sources/views/contacts.js

```diff
@@ -1,43 +1,83 @@
 import { JetView } from "webix-jet";
 import { contacts } from "models/contacts";
 
-import Form from "views/form";
-
 export default class Contacts extends JetView {
 	config() {
-		const list = {
-			padding: {
-				right: 10
-			},
-			rows: [
+		const _ = this.app.getService("locale")._;
+		return {
+			cols: [
 				{
-					type: "header",
-					template: "Contacts",
-					height: 50,
-					css: "webix_dark"
+					padding: {
+						right: 10
+					},
+					rows: [
+						{
+							type: "header",
+							template: _("Contacts"),
+							height: 50,
+							css: "webix_dark"
+						},
+						{
+							view: "list",
+							localId: "contactsList",
+							autowidth: true,
+							scroll: false,
+							select: true,
+							type: {
+								height: 62
+							},
+							template: `
+								<b>${_("Name")}: </b>#Name# 
+								<span class="webix_icon wxi-close remove_icon" style="float: right"></span> 
+								<br> 
+								<b>${_("Email")}: </b>#Email# `,
+							onClick: {
+								"wxi-close": (e, id) => this.deleteItem(id)
+							},
+						},
+						{
+							view: "button",
+							label: _("Add"),
+							align: "right",
+							autoWidth: true,
+							css: "webix_primary",
+							click: () => this.addItem()
+						},
+
+					]
 				},
-				{
-					view: "list",
-					localId: "contactsList",
-					autowidth: true,
-					scroll: false,
-					select: true,
-					template: "#Name# <br> #Email# <span class='webix_icon wxi-close right'></span>"
-				}
+				{ $subview: true }
 			]
 		};
+	}
+	init(view, url) {
+		this.list = this.$$("contactsList");
+		this.list.sync(contacts);
+		this.list.attachEvent("onItemClick", id => {
+			this.show(`./form?id=${id}`);
+		});
 
-		const ui = {
-			cols: [
-				list,
-				Form
-			]
-		};
+		if (url.length <= 1) {
+			this.show(`./form?id=${this.list.getFirstId()}`);
```

Comment:
(1) what if there are not contacts in the list? 

(2) what if I write an incorrect id `top/contacts/form?id=25` in the address bar?

Reply:
(1)(2) Added additional checking in ternary operators.